includes/class-amp-theme-support.php

```diff
@@ -154,6 +154,8 @@ public static function register_paired_hooks() {
 	 */
 	public static function register_hooks() {
 
+		add_action( 'widgets_init', array( __CLASS__, 'register_widgets' ) );
```

Comment:
Hi @westonruter, 
It looks like '`widgets_init'` has already fired by the time `register_hooks()` executes. 

`AMP_Theme_Support::init()` [is called on the 'wp' action](https://github.com/Automattic/amp-wp/blob/develop/amp.php#L137), which is after `'widgets_init'`.

My suggestion of checking `is_amp_endpoint()` probably led to this. That function [needs to be called](https://github.com/Automattic/amp-wp/blob/develop/includes/amp-helper-functions.php#L126) before `'parse_query'`, which is already after `'widgets_init'`.

Reply:
@kienstra great point. It looks like maybe the theme-support logic needs to be removed from `amp_maybe_add_actions()` to instead be called earlier? But yeah, tough spot with the dependency on the `parse_query` action.

Reply:
**Moved register_widgets()**

Hi @westonruter,
What do you think of [this commit](https://github.com/Automattic/amp-wp/pull/870/commits/6053f9a0d4a4fff51a423845ecd28e86903d1988), which ensures that `register_widgets()` fires in the right place?